src/r3f/TilesRendererComponent.jsx

```diff
@@ -0,0 +1,237 @@
+
+	const tokens = [ ...path ];
+	const finalKey = tokens.pop();
+	getValueAtPath( object, tokens )[ finalKey ] = value;
+
+}
+
+// context for accessing the tile set
+export const TilesRendererContext = createContext( null );
+
+// component for registering a plugin
+export function TilesPluginComponent( props ) {
+
+	const { plugin, args, ...options } = props;
+	const tiles = useContext( TilesRendererContext );
+	const [ instance, setInstance ] = useState( null );
+
+	// creates the instance
+	useEffect( () => {
+
+		if ( tiles === null ) {
+
+			return;
+
+		}
+
+		let instance;
+		if ( Array.isArray( args ) ) {
+
+			instance = new plugin( ...args );
+
+		} else {
+
+			instance = new plugin( args );
+
+		}
+
+		tiles.registerPlugin( instance );
+		setInstance( instance );
+
+		return () => {
+
+			tiles.unregisterPlugin( instance );
+
+		};
+
+	}, [ plugin, tiles, ...getDepsArray( args ) ] );
+
+	// assigns any provided options to the plugin
+	useEffect( () => {
+
+		if ( instance === null ) {
+
+			return;
+
+		}
+
+		for ( const key in options ) {
+
+			if ( key in instance ) {
+
+				instance[ key ] = options[ key ];
+
+			}
+
+		}
+
+	}, [ instance, ...getDepsArray( options ) ] );
+
+}
+
+// component for adding a TilesRenderer to the scene
+export function TilesRendererComponent( props ) {
+
+	const { url, children, ...options } = props;
+	const [ tiles, setTiles ] = useState( null );
+	const [ camera, gl ] = useThree( state => [ state.camera, state.gl ] );
+
+	// create the tile set
+	useEffect( () => {
+
+		const tiles = new TilesRenderer( url );
+		setTiles( tiles );
+
+		return () => {
+
+			tiles.dispose();
+
+		};
+
+	}, [ url ] );
+
+	// update the resolution for the camera
+	useFrame( () => {
+
+		camera.updateMatrixWorld();
+		tiles.setResolutionFromRenderer( camera, gl );
+		tiles.update();
+
+	} );
+
+	// add the camera
+	useEffect( () => {
+
+		if ( tiles === null ) {
+
+			return;
+
+		}
+
+		tiles.setCamera( camera );
+		return () => {
+
+			tiles.deleteCamera( camera );
+
+		};
+
+	}, [ tiles, camera ] );
+
+	// assign options recursively
+	useEffect( () => {
+
+		if ( tiles === null ) {
+
+			return;
+
+		}
+
+		let previousState = {};
+		for ( const key in options ) {
+
+			if ( isEventName( key ) ) {
+
+				tiles.addEventListener( getEventName( key ), options[ key ] );
+
+			} else {
+
+				const path = getPath( key );
+				previousState[ key ] = getValueAtPath( tiles, path );
+				setValueAtPath( tiles, path, options[ key ] );
+
+			}
+
+		}
+
+		return () => {
+
+			for ( const key in options ) {
+
+				if ( isEventName( key ) ) {
+
+					tiles.removeEventListener( getEventName( key ), options[ key ] );
+
+				}
+
+			}
+
+			for ( const key in options ) {
+
+				const path = getPath( key );
+				setValueAtPath( tiles, path, options[ key ] );
+
+			}
+
+		};
+
+	}, [ tiles, ...getDepsArray( options ) ] );
+
+	return <>
+		{ tiles ? <primitive object={ tiles.group }/> : null }
```

Comment:
One thing I don't love is that we can't pass transformation parameters to the tiles group (like `position`, `scale-x`, etc) because of how options are handled implicitly on the TilesRenderer class. Maybe there's another way to handle that?

Reply:
Should not these options be left at the discretion of the user that embed the component? The same way you defined a rotation on a `group` that encapsulates the `TilesRendererComponent ` here. Since these matrix transforms cascade, from my point of view this should be the same. 
```jsx
	<group rotation-x={ Math.PI / 2 }>
		<TilesRendererComponent url={ URL } lruCache-minSize={ 0 }>
			...
		</TilesRendererComponent>
	</group>
```

Reply:
Yeah it's ultimately the same - just an extra unnecessary object / matrix calculation that's required but you're right it's probably not a big deal